cmd/corimDisplay.go

```diff
@@ -61,72 +61,119 @@ func checkCorimDisplayArgs() error {
-	var (
-		signedCorimCBOR []byte
-		metaJSON        []byte
-		corimJSON       []byte
-		err             error
-		s               corim.SignedCorim
-	)
-
-	if signedCorimCBOR, err = afero.ReadFile(fs, signedCorimFile); err != nil {
-		return fmt.Errorf("error loading signed CoRIM from %s: %w", signedCorimFile, err)
+func displaySignedCorim(s corim.SignedCorim, corimFile string, showTags bool) error {
+	metaJSON, err := json.MarshalIndent(&s.Meta, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error encoding CoRIM Meta from %s: %w", corimFile, err)
 	}
 
-	if err = s.FromCOSE(signedCorimCBOR); err != nil {
-		return fmt.Errorf("error decoding signed CoRIM from %s: %w", signedCorimFile, err)
+	fmt.Println("Meta:")
+	fmt.Println(string(metaJSON))
+
+	corimJSON, err := json.MarshalIndent(&s.UnsignedCorim, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error encoding unsigned CoRIM from %s: %w", corimFile, err)
 	}
 
-	if metaJSON, err = json.MarshalIndent(&s.Meta, "", "  "); err != nil {
-		return fmt.Errorf("error decoding CoRIM Meta from %s: %w", signedCorimFile, err)
+	fmt.Println("Corim:")
+	fmt.Println(string(corimJSON))
+
+	if showTags {
+		fmt.Println("Tags:")
+		// convert []corim.Tag to [][]byte
+		tags := make([][]byte, len(s.UnsignedCorim.Tags))
+		for i, tag := range s.UnsignedCorim.Tags {
+			tags[i] = tag
+		}
+		displayTags(tags)
 	}
 
-	fmt.Println("Meta:")
-	fmt.Println(string(metaJSON))
+	return nil
+}
 
-	if corimJSON, err = json.MarshalIndent(&s.UnsignedCorim, "", "  "); err != nil {
-		return fmt.Errorf("error decoding unsigned CoRIM from %s: %w", signedCorimFile, err)
+func displayUnsignedCorim(u corim.UnsignedCorim, corimFile string, showTags bool) error {
+	corimJSON, err := json.MarshalIndent(&u, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error encoding unsigned CoRIM from %s: %w", corimFile, err)
 	}
 
 	fmt.Println("Corim:")
 	fmt.Println(string(corimJSON))
 
 	if showTags {
 		fmt.Println("Tags:")
-		for i, e := range s.UnsignedCorim.Tags {
-			// need at least 3 bytes for the tag and 1 for the smallest bstr
-			if len(e) < 3+1 {
-				fmt.Printf(">> skipping malformed tag at index %d\n", i)
-				continue
-			}
-
-			// split tag from data
-			cborTag, cborData := e[:3], e[3:]
-
-			hdr := fmt.Sprintf(">> [ %d ]", i)
-
-			if bytes.Equal(cborTag, corim.ComidTag) {
-				if err = printComid(cborData, hdr); err != nil {
-					fmt.Printf(">> skipping malformed CoMID tag at index %d: %v\n", i, err)
-				}
-			} else if bytes.Equal(cborTag, corim.CoswidTag) {
-				if err = printCoswid(cborData, hdr); err != nil {
-					fmt.Printf(">> skipping malformed CoSWID tag at index %d: %v\n", i, err)
-				}
-			} else if bytes.Equal(cborTag, cots.CotsTag) {
-				if err = printCots(cborData, hdr); err != nil {
-					fmt.Printf(">> skipping malformed CoTS tag at index %d: %v\n", i, err)
-				}
-			} else {
-				fmt.Printf(">> unmatched CBOR tag: %x\n", cborTag)
-			}
+		// convert []corim.Tag to [][]byte
```

Comment:
A translation here is duplicative.

Reply:
i will add this code to display function itself